src/Orleans.Serialization/Codecs/IntegerCodec.cs

```diff
@@ -745,4 +749,210 @@ public static ulong ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+
+        /// <inheritdoc/>
+        void IFieldCodec<UInt128>.WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, UInt128 value) => WriteField(ref writer, fieldIdDelta, expectedType, value);
+
+        /// <summary>
+        /// Writes a field.
+        /// </summary>
+        /// <typeparam name="TBufferWriter">The buffer writer type.</typeparam>
+        /// <param name="writer">The writer.</param>
+        /// <param name="fieldIdDelta">The field identifier delta.</param>
+        /// <param name="expectedType">The expected type.</param>
+        /// <param name="value">The value.</param>
+        public static void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, UInt128 value) where TBufferWriter : IBufferWriter<byte>
+        {
+            if (value <= (UInt128)ulong.MaxValue && value >= (UInt128)ulong.MinValue)
+            {
+                UInt64Codec.WriteField(ref writer, fieldIdDelta, expectedType, (ulong)value, CodecFieldType);
+            }
+            else
+            {
+                ReferenceCodec.MarkValueField(writer.Session);
+                UnsignedBinaryIntegerCodec<UInt128>.WriteField(ref writer, fieldIdDelta, expectedType, in value);
+            }
+        }
+
+        /// <inheritdoc/>
+        UInt128 IFieldCodec<UInt128>.ReadValue<TInput>(ref Reader<TInput> reader, Field field) => ReadValue(ref reader, field);
+
+        /// <summary>
+        /// Reads a value.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <param name="field">The field.</param>
+        /// <returns>The value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static UInt128 ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        {
+            if (field.WireType != WireType.LengthPrefixed)
+            {
+                return UInt64Codec.ReadValue(ref reader, field);
+            }
+
+            ReferenceCodec.MarkValueField(reader.Session);
+            UInt128 result;
+            UnsignedBinaryIntegerCodec<UInt128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Reads a value without protocol framing.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <returns>The value.</returns>
+        internal static UInt128 ReadRaw<TInput>(ref Reader<TInput> reader)
+        {
+            UInt128 result;
+            UnsignedBinaryIntegerCodec<UInt128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+    }
+
+    internal static class SignedBinaryIntegerCodec<T> where T : struct, IBinaryInteger<T>, ISignedNumber<T>
```

Comment:
Is this used anywhere else (besides `Int128`)?

Reply:
It's just there. I'll in-line it

Reply:
This code unfortunately seems to allocate. Inlining it also results in allocations for the boxed `Int128`. Perhaps it's an issue with static interface methods or perhaps I'm doing something obviously wrong. If you spot it, please feel free to either open a subsequent PR, push a commit to this branch, or point it out